client/src/main/java/io/pravega/client/security/auth/JwtTokenHandlingStrategy.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client.security.auth;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import io.pravega.client.stream.impl.Controller;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import io.pravega.common.util.ConfigurationOptionsExtractor;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.Base64;
+import java.util.concurrent.atomic.AtomicReference;
+
+@Slf4j
+public class JwtTokenHandlingStrategy implements DelegationTokenHandlingStrategy {
+
+    /**
+     * Represents the threshold for triggering delegation token refresh.
+     */
+    @VisibleForTesting
+    static final int DEFAULT_REFRESH_THRESHOLD = 5;
+
+    @Getter(AccessLevel.PROTECTED)
+    private static final int REFRESH_THRESHOLD;
+
+    /**
+     * The Controller gRPC client used for interacting with the server.
+     */
+    private final Controller controllerClient;
+
+    private final String scopeName;
+
+    private final String streamName;
+
+    @Getter(AccessLevel.PROTECTED)
+    private AtomicReference<DelegationToken> delegationToken = new AtomicReference<>();
+
+    static {
+        REFRESH_THRESHOLD = ConfigurationOptionsExtractor.extractInt(
```

Comment:
This is not following dependency injection. This will be very difficult to test. Consider if there is a way this can be provided via a config object, through construction, or obtained from the controller.

Reply:
Sure, I've made it an instance variable now and have added overloaded constructors so that the configuration may be injected via the constructor for testing purposes.